### PR TITLE
chore(docs): update quickstart.md to use YAML

### PR DESCRIPTION
**Pull request by pront**
part of #18323

### CODE REVIEW

File: website/content/en/docs/setup/quickstart.md [hunk 1]
```diff
@@ -165,15 +209,27 @@ We're just scratching the surface in this post. To get your hands dirty with Vec
 * Finally, [deploying Vector][deployment] to launch Vector in your production environment.
 
 [components]: /components
+
```
Comment (jszwedko) [inline website/content/en/docs/setup/quickstart.md:212]: Did you mean to add all of these extra blank lines?

**Review by jszwedko — APPROVED** [review]
Thanks!

File: website/content/en/docs/setup/quickstart.md [hunk 2]
```diff
@@ -1,14 +1,16 @@
 ---
-title: Vector quickstart
+title:       Vector quickstart
```
Comment (pront) [inline website/content/en/docs/setup/quickstart.md:2]: This was auto-formatted. If it's too distracting, I can revert the formatting changes.

File: website/content/en/docs/setup/quickstart.md (hunk 1 above)
Comment (pront) [reply website/content/en/docs/setup/quickstart.md]: Reverted all the formatting changes.

**Review by jszwedko — APPROVED** [review]
👍 on just making the content changes. It might be nice to introduce a standard markdown formatter to our dev workflow.

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**